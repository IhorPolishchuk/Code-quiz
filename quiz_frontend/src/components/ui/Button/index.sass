$button_color: #9680aa
$text_color: #ffffff

.button
    display: flex
    align-items: center
    padding: 5px 15px
    border-radius: 15px
    background: $button_color
    color: $text_color
    transition: all .35s 
    cursor: pointer
    &:hover
        background: darken($button_color, 15%)