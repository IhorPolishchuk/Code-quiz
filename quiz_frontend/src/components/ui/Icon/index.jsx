import React from 'react';
import PropTypes from 'prop-types';

/**
 * Icon component
 * 
 * @prop {string} icon    : icon path, required
 * @prop {number} size    : icon size, by default is 20
 * @prop {string} color   : icon color, by default is #ffffff
 * @prop {string} bgColor : icon background color, by default is rgba(0, 0, 0, 0)
 * 
 * @returns 
 */
const Icon = props => {
  const styles = {
    svg: {
      display: 'inline-block',
      verticalAlign: 'middle',
      backgroundColor: props.bgColor
    },
    path: {
      fill: props.color,
    },
  };

  return (
    <svg
      style={{...styles.svg, ...props.style}}
      width={`${props.size}px`}
      height={`${props.size}px`}
      viewBox="0 0 1024 1024"
    >
      <path
        style={styles.path}
        d={props.icon}
      />
    </svg>
  );
};

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  size: PropTypes.number,
  color: PropTypes.string,
  bgColor: PropTypes.string
};

Icon.defaultProps = {
  size: 20,
  bgColor: 'rgba(0,0,0,0)',
  color: '#ffffff'
};

export default Icon;
