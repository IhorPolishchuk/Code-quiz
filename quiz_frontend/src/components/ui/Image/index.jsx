import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Image extends Component {
    render () {
        const { src, alt, width, height, className } = this.props;
        const style = width && height ? {width: `${width}px`, height: `${height}px`} : {};
        return (
            <img 
                src={src} 
                alt={alt} 
                className={className}
                style={style}
            />
        );
    }
}

Image.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string,
    className: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number
};

Image.defaultProps = {
    alt: '',
    className: '',
};

export default Image;
