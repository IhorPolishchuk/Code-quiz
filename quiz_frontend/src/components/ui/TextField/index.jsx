import React, { Component } from 'react';
import PropTypes from 'prop-types';
/**
 * ToDo: Styles and Tests for TextField
 */
class TextField extends Component {
    constructor(props){
        super(props);
        this.state = {
            value: this.props.value
        };
        this.onChange = this.onChange.bind(this);
    }

    onChange(element){
        this.setState({
            value: element.target.value
        });
        if(this.props.onChange){
            this.props.onChange(element);
        }   
    }

    render () {
        const { type, label } = this.props;
        const { value } =this.state;
        const id = `${new Date().getTime()}_${label}}`;
        return (
            <div>
                <label htmlFor={id}>
                    {label}
                </label>
                <input 
                    id={id} 
                    type={type} 
                    value={value} 
                    onChange={this.onChange}
                /> 
            </div>
        );
    }
}

TextField.propTypes = {
    type: PropTypes.oneOf['text', 'password'],
    label: PropTypes.string.isRequired,
    value: PropTypes.string
};

TextField.defaultProps = {
    type: 'text',
    value: ''
};

export default TextField;
