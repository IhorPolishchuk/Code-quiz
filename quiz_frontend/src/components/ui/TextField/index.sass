.label
    position: absolute
    top: 35px
    display: block
    font-size: 16px
    color: #e2e2e2
    transition: all 0.5s

.input
    position: relative
    background: transparent
    width: 100%
    border: none
    outline: none
    padding: 8px 0
    font-size: 16px

.wrapper
    padding: 30px 0 0 0
    position: relative

.wrapper:after, .wrapper:before
    content: ""
    position: absolute
    display: block
    width: 100%
    height: 2px
    background-color: #e2e2e2
    bottom: 0
    left: 0
    transition: all 0.5s

.wrapper::after
    background-color: #8E8DBE
    transform: scaleX(0)

.active::after
    transform: scaleX(1)

.active .label
    color: #8E8DBE

.completed .label
    font-size: 14px
    top: 20px
