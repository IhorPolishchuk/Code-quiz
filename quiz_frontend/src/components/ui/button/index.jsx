import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Icon } from './../index.js';

import styles from './index.sass';

class Button extends Component {
    constructor(props){
        super(props);
        this.state = {
            iconLeft: this.props.iconLeft,
            iconRight: this.props.iconRight
        };
    }

    
    componentWillMount (){
        const { iconLeft, iconRight } = this.props;
        switch (true) {
            case (!iconLeft && iconRight):
                this.setState({
                    iconLeft: false,
                    iconRight: true                    
                });
                break;
            case (iconLeft && !iconRight):
                this.setState({
                    iconLeft: true,
                    iconRight: false                    
                });                
                break;
            case (iconLeft && iconRight) || (!iconLeft && !iconRight):
            default:
                this.setState({
                    iconLeft: true,
                    iconRight: false               
                });
                console.warn('Params iconLeft and iconRight should not be the same.');
                break;
        }
    }

    render () {
        const { title, icon, iconColor,...rest } = this.props;
        const { iconLeft, iconRight } = this.state;
        return (
            <div
                className={styles.button}
                icon={icon}
                {...rest}
            >   
                {icon && iconLeft &&
                    <Icon
                        icon={icon}
                        iconColor={iconColor}
                    />
                }
                {title}
                {icon && iconRight &&
                    <Icon
                        icon={icon}
                        color={iconColor}
                    />
                }
            </div>
        );
    }
}

Button.propTypes = {
    title: PropTypes.string,
    icon: PropTypes.string,
    iconLeft: PropTypes.bool,
    iconRight: PropTypes.bool,
    iconColor: PropTypes.string
};

Button.defaultProps = {
    title: 'Button',
    iconLeft: true,
    iconRight: false,
    iconColor: '#ffffff'
};

export default Button;
