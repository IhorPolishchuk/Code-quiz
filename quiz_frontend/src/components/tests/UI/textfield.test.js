import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { TextField } from './../../ui';

describe('TextField component test', () => {
    it('TextField renders with default props', () => {
        const textfield = shallow(
            <TextField
                type="text"
                label="text"
            />
        );
        const tree = toJson(textfield);
        
        expect(tree).toMatchSnapshot();

    }); 
});
