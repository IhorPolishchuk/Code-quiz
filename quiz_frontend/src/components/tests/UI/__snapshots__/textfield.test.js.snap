// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField component test TextField renders with default props 1`] = `
<div
  className="  
                wrapper 
                
                
            "
>
  <label
    className="label"
    htmlFor="text"
  >
    text
  </label>
  <input
    className="input"
    id="text"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;
